packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -100,1337 +29,28 @@ class UpdatePackagesCommand extends FlutterCommand {
-        await pub.batch(
-          <String>['deps', '--style=compact'],
-          context: PubContext.updatePackages,
-          directory: tempDir.path,
-          filter: tree.fill,
-          retry: false, // errors here are usually fatal since we're not hitting the network
-        );
-      } finally {
-        tempDir.deleteSync(recursive: true);
-      }
-
-      // The transitive dependency tree for the fake package does not contain
-      // dependencies between Flutter SDK packages and pub packages. We add them
-      // here.
-      for (final PubspecYaml pubspec in pubspecs) {
-        final String package = pubspec.name;
-        specialDependencies.add(package);
-        tree._versions[package] = pubspec.version;
-        assert(!tree._dependencyTree.containsKey(package));
-        tree._dependencyTree[package] = <String>{};
-        for (final PubspecDependency dependency in pubspec.dependencies) {
-          if (dependency.kind == DependencyKind.normal) {
-            tree._dependencyTree[package].add(dependency.name);
-          }
-        }
-      }
-
-      if (isPrintTransitiveClosure) {
-        tree._dependencyTree.forEach((String from, Set<String> to) {
-          globals.printStatus('$from -> $to');
-        });
-        return FlutterCommandResult.success();
-      }
-
-      if (isPrintPaths) {
-        showDependencyPaths(from: stringArg('from'), to: stringArg('to'), tree: tree);
-        return FlutterCommandResult.success();
-      }
-
-      // Now that we have collected all the data, we can apply our dependency
-      // versions to each pubspec.yaml that we collected. This mutates the
-      // pubspec.yaml files.
-      //
-      // The specialDependencies argument is the set of package names to not pin
-      // to specific versions because they are explicitly pinned by their
-      // constraints. Here we list the names we earlier established we didn't
-      // need to pin because they come from the Dart or Flutter SDKs.
-      for (final PubspecYaml pubspec in pubspecs) {
-        pubspec.apply(tree, specialDependencies);
-      }
-
-      // Now that the pubspec.yamls are updated, we run "pub get" on each one so
-      // that the various packages are ready to use. This is what "flutter
-      // update-packages" does without --force-upgrade, so we can just fall into
-      // the regular code path.
-    }
-
-    final Stopwatch timer = Stopwatch()..start();
-    int count = 0;
-
-    for (final Directory dir in packages) {
-      await pub.get(
-        context: PubContext.updatePackages,
-        directory: dir.path,
-        checkLastModified: false,
-        offline: offline,
-      );
-      count += 1;
-    }
-
-    await _downloadCoverageData();
-
-    final double seconds = timer.elapsedMilliseconds / 1000.0;
-    globals.printStatus("\nRan 'pub' $count time${count == 1 ? "" : "s"} and fetched coverage data in ${seconds.toStringAsFixed(1)}s.");
-
+    await globals.processManager.run(
```

Comment:
Should this command print an error message that it is probably not interesting/useful to end-users?

Reply:
its already hidden, but I added a comment that it is being moved and what the new command is